test/consistent-package-json.test.ts

```diff
@@ -0,0 +1,135 @@
+import {readJSONSync} from 'fs-extra';
+import {dirname, join, relative, resolve} from 'path';
+import glob from 'glob';
+
+const KNOWN_TEMPLATE_KEYS = [
+  'author',
+  'bugs',
+  'dependencies',
+  'description',
+  'files',
+  'homepage',
+  'license',
+  'main',
+  'name',
+  'publishConfig',
+  'repository',
+  'scripts',
+  'sideEffects',
+  'types',
+  'version',
+];
+
+const ROOT_PATH = resolve(__dirname, '..');
+const packageJSONTemplatePath = join('templates', 'package.hbs.json');
+const packageJSONTemplate = readJSONSync(join(ROOT_PATH, packageJSONTemplatePath));
+
+describe(packageJSONTemplatePath, () => {
+  it('has only known keys', () => {
+    const keys = Object.keys(packageJSONTemplate).sort();
+
+    // Template keys should match exactly. If updating it, remember to update tests!
+    expect(keys).toEqual(KNOWN_TEMPLATE_KEYS);
+  });
+});
+
+readPackages().forEach(({packageName, packageJSON, packageJSONPath}) => {
+  describe(packageJSONPath, () => {
+    it('specifies Shopify as author', () => {
+      expect(packageJSON.author).toBe(packageJSONTemplate.author);
+    });
+
+    it('specifies Quilt Issues as bugs URL', () => {
+      expect(packageJSON.bugs).toEqual(packageJSONTemplate.bugs);
+    });
+
+    it('specifies a dependencies', () => {
+      expect(packageJSON.dependencies).not.toEqual({});
+    });
+
+    it('specifies a description', () => {
+      expect(packageJSON.description).not.toBeUndefined();
+    });
+
+    it('specifies files', () => {
+      if(packageName == 'polyfills') return; // FIXME
+
+      expect(packageJSON.files).toContain('dist/*');
+      expect(packageJSON.files).toContain('!tsconfig.tsbuildinfo');
+    });
+
+    it('specifies Quilt deep-link homepage', () => {
+      const expectedHomepage = compile(packageJSONTemplate.homepage);
+
+      expect(packageJSON.homepage).toBe(expectedHomepage);
+    });
+
+    it('specifies the MIT license', () => {
+      expect(packageJSON.license).toBe(packageJSONTemplate.license);
+    });
+
+    it('specifies the expected main', () => {
+      if(packageName === 'graphql-persisted') return; // FIXME: Address this in graphql-persisted
```

Comment:
The `graphql-persisted` package does not specify an entry for `main`.

Reply:
Would it not still make sense to have a `main` `index.js` that consists of something like

```javascript
export {apollo} from './apollo';
export {apollo} from './koa';
```

This would make it conform to the expected main entry point, while still offering the option to import the desired consumer directly. As a side benefit, it would also allow the `types` entry to be consistent.